### PR TITLE
Bluetooth: host: Fix undefined references to adv in central only builds 

### PR DESCRIPTION
**Pull request by joerchan**
Fix regression is central only builds when CONFIG_BT_PRIVACY and/or
CONFIG_BT_EXT_ADV is enabled, or the functions bt_id_reset or
bt_id_delete is called from application.

This resulted in build error for undefined functions
bt_le_ext_adv_foreach and bt_le_adv_lookup_legacy since the definition
for these functions are now only compiled in when CONFIG_BT_BROADCASTER
has been enabled.

Regression from:
53cea47

Add privacy configuration to init test for both central and peripheral
roles with or without extended advertising.

### CODE REVIEW

**Review by jhedberg — APPROVED** [review]

File: tests/bluetooth/init/prj_ctlr_peripheral_priv.conf [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+CONFIG_BT_BROADCASTER=y
+CONFIG_BT_PERIPHERAL=y
+CONFIG_BT_OBSERVER=n
+CONFIG_BT_CENTRAL=n
+CONFIG_BT_SMP=y
+CONFIG_BT_PRIVACY=y
+CONFIG_BT_EXT_ADV=n
+CONFIG_BT_CTLR_ADV_EXT=n
+CONFIG_BT_LL_SW_SPLIT=y
+CONFIG_ZTEST=y
```
Comment (trond-snekvik) [inline tests/bluetooth/init/prj_ctlr_peripheral_priv.conf:12]: These files need entries in testcase.yaml to get included in CI

File: tests/bluetooth/init/prj_ctlr_central priv.conf [hunk 2]
```diff
@@ -0,0 +1,12 @@
+CONFIG_BT=y
```
Comment (trond-snekvik) [inline tests/bluetooth/init/prj_ctlr_central priv.conf:1]: Use underscore instead of space in filename

**Review by trond-snekvik — CHANGES_REQUESTED** [review]

**Review by Thalley — APPROVED** [review]
LGTM

File: tests/bluetooth/init/prj_ctlr_peripheral_priv.conf (hunk 1 above)
Comment (joerchan) [reply tests/bluetooth/init/prj_ctlr_peripheral_priv.conf:12]: Updated testcase.yaml, thank you.

File: tests/bluetooth/init/prj_ctlr_central priv.conf (hunk 2 above)
Comment (joerchan) [reply tests/bluetooth/init/prj_ctlr_central priv.conf]: must be something wrong with my keyboard :P

**Review by trond-snekvik — APPROVED** [review]

**Comment by jhedberg** [comment]
@joerchan there seems to be a build failure in one of the Bluetooth tests

**Comment by jhedberg** [comment]
> @joerchan there seems to be a build failure in one of the Bluetooth tests

Answering myself: looks like it might have gotten fixed by #33412 which is now merged, so I retriggered the build.

**Pull request merged**